LocalGround/localground
# Upload media

### Summary of Changes
This commit took on a life of it's own as fixes lead to other fixes (not ideal, I know), but to summarize...
* Uploader UI: 
  * old Django endpoints have been deprecated
  * moved the file upload post-processing functionality from the models to the serializers so that more validation could happen before an insert / update is issued
  * UI simplified so that user doesn't have to make the photo / audio distinction
  * Looks better (plus upgraded to newer bootstrap library...was using legacy one)
  * iFrame version of UI (for map editor) is simpler / smaller
  * better error handling
* API: 
   * deprecated the AuditCreate and AuditUpdate mixins, as that functionality made much more sense in the serializer (and simplified the code significantly)
  * fixed some issues with file paths for the photo and audio file uploader endpoint and updated tests
  * updated the scan uploader endpoint to accept file uploads
  * added the ability to upload files through the API browser for photos, audio files, and scans
  * made sure that the file upload is not available for PUT (update) requests

### Testing instructions
Please check the following things:
* REST API
  1. Can you successfully add photos, audios, and scans through the REST API (/api/0/(photos, audio, map-images)?
  2. Does it give you a graceful error if you try to upload the wrong file type?
  3. Does it not offer you the file upload option for existing media records?
* Upload UI
   1.  i. & ii. above
   2. any changes to the UI?
* Map Editor
   1. i. & ii. above
   2. When you close the modal, does the map update?
